<launch>

    <arg name="port" default="/dev/ttyS0"/>
    <arg name="baud" default="115200"/>
    <arg name="test_file" default=""/>
    <arg name="sensor_hosts" default="none"/>
    
    <include file="$(find htp_auto)/launch/record.launch" />

    <node pkg="nmea_navsat_driver" type="nmea_serial_driver" name="gps_driver">
        <param name="port" value="$(arg port)"/>
        <param name="baud" value="$(arg baud)"/>
        <param name="test_file" value="$(arg test_file)"/>
    </node>

    <node pkg="htp_auto" type="gps_converter" name="gps_converter"> </node>
    
    <node pkg="htp_auto" type="sensor_control_client.py" name="sensor_control_client">
        <param name="sensor_hosts" value="$(arg sensor_hosts)"/>
    </node>

    <node name="gps_utc_throttler" type="throttle" pkg="topic_tools" args="messages /gps_utc 20 /gps_utc_throttled" />

</launch>
